src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/DefaultEditorTemplates.cs

```diff
@@ -378,6 +378,54 @@ public static IHtmlContent MonthInputTemplate(IHtmlHelper htmlHelper)
             return GenerateTextBox(htmlHelper, inputType: "month");
         }
 
+        public static IHtmlContent WeekInputTemplate(IHtmlHelper htmlHelper)
```

Comment:
Align the `InputTagHelper` and `Html.TextBox(...)` approaches. Do not place the same 30 lines in two places.

Reply:
For example, might set `FormattedModelValue` in `TemplateBuilder` and `modelExplorer.Model` (passed to `IHtmlGenerator`) in `InputTagHelper` using a shared helper method you stick somewhere accessible to both.

The helper method itself needs unit tests. Must test with more than a single input value e.g. land in the 53rd week of a year.

Reply:
Can the `FormattedModelValue` be set in `DefaultEditorTemplates` instead? What's the value of passing it to `TemplateBuilder`? Tag helpers do not use the `TemplateBuilder` right?

Reply:
The reason to set `FormattedModelValue` in `TemplateBuilder` is that's where it's otherwise set. You'll still need to add a new template for `week` to set the element's `type` correctly.